packages/payment-detection/src/any/any-to-erc20-proxy-info-retriever.ts

```diff
@@ -101,63 +116,69 @@ export default class ProxyERC20InfoRetriever
     // Get the fee proxy contract event logs
     const feeLogs = await this.provider.getLogs(feeFilter);
 
+    const logToObject = <T>({ args, eventFragment }: LogDescription): T => {
+      return args.reduce((prev, current, i) => {
+        prev[eventFragment.inputs[i].name] = current;
+        return prev;
+      }, {});
+    };
     // Parses, filters and creates the events from the logs with the payment reference
     const eventPromises = conversionLogs
       // Parses the logs
       .map((log) => {
         const parsedConversionLog = this.contractConversionProxy.interface.parseLog(log);
-
         const proxyLog = feeLogs.find((l) => l.transactionHash === log.transactionHash);
-        let parsedProxyLog;
-        if (proxyLog) {
-          parsedProxyLog = this.contractERC20FeeProxy.interface.parseLog(proxyLog);
+        if (!proxyLog) {
+          throw new Error('proxy log not found');
         }
```

Comment:
@vrolland is this OK? later in the code, we consider this to never be null, so I assume there should be an error here